src/mmda/types/span.py

```diff
@@ -53,3 +53,16 @@ def small_spans_to_big_span(cls, spans: List["Span"]) -> "Span":
             end=end,
             box=Box.small_boxes_to_big_box(boxes=[span.box for span in spans]),
         )
+
+    def is_overlap(self, other: "Span") -> bool:
+        is_self_before_other = self.start < other.end and self.end > other.start
+        is_other_before_self = other.start < self.end and other.end > self.start
```

Comment:
You know, I think I misread this the first time and your original condition was sufficient, i.e.

`self.start < other.end and self.end > other.start`